examples/sharing_asset.py

```diff
@@ -0,0 +1,191 @@
+   access rights.
+"""
+
+from json import dumps as json_dumps
+from os import getenv
+
+from archivist.archivist import Archivist
+from archivist.constants import ASSET_BEHAVIOURS, SUBJECTS_SELF_ID
+from archivist.logger import set_logger
+from archivist.proof_mechanism import ProofMechanism
+from archivist.utils import get_auth
+
+
+def create_asset(arch, label):
+    """Create an asset using Archivist Connection.
+
+    Args:
+        arch: archivist connection.
+        label: convenience label to easily distinguish the 2 organizations.
+
+    Returns:
+        Asset: a new asset created.
+
+    """
+    attrs = {
+        "arc_display_name": f"{label}_display_name",  # Asset's display name
+        "arc_description": f"{label}_display_description",  # Asset's description
+        "arc_display_type": f"{label}_display_type",  # Arc_display_type is a free text field
+        "ext_vendor_name": label,
+    }
+
+    # Select the mechanism used to prove evidence for the asset.  If the selected proof
+    # mechanism is not enabled for your tenant then an error will occur.
+    # If unspecified then SIMPLE_HASH is used.
+    # proof_mechanism = ProofMechanism.KHIPU.name
+    #
+    props = {
+        "proof_mechanism": ProofMechanism.KHIPU.name,
+    }
+
+    # The first argument is the properties of the asset
+    # The second argument is the attributes of the asset
+    # The third argument is wait for confirmation:
+    #   If @confirm@ is True then this function will not
+    #   return until the asset is confirmed on the blockchain and ready
+    #   to accept events (or an error occurs)
+    #
+    return arch.assets.create(props=props, attrs=attrs, confirm=True)
+
+
+def create_archivist(label):
+    """Create connection to archivist"""
+    # Get authorization token. The token grants certain rights and access permissions.
+    # The token can represent the root principal or user in an organization. Different tokens
+    # could indicate different users in the same organization or membership of different
+    # organiastions.
+    auth = get_auth(
+        auth_token=getenv(f"ARCHIVIST_AUTHTOKEN_{label}"),
+    )
+    # Initialize connection to Archivist. max_time is the time to wait for confirmation
+    # of an asset or event creation - the default is 1200 seconds but one can optionally
+    # specify a different value here particularly when creating assets on SIMPLE_HASH
+    # (rather than KHIPU) as confirmation times are much shorter in this case.
+    return Archivist(
+        "https://app.rkvst.io",
+        auth,
+        max_time=300,
+    )
+
+
+def exchange_subject(acme, weyland):
+    """Add subjects record for weyland in acme's environment"""
+    subject = acme.subjects.exchange(
+        "weyland",
+        weyland.subjects.read(SUBJECTS_SELF_ID),
+    )
+
+    # must wait for confirmation
+    acme.subjects.wait_for_confirmation(subject["identity"])
+
+    return subject
+
+
+def create_access_policy(arch, label, subject):
```

Comment:
Again - I think the example code suffers in readability because this looks like a general purpose function - but its actually accomplishing a specific task.